camus-etl-kafka/src/main/java/com/linkedin/camus/etl/kafka/mapred/EtlMultiOutputRecordWriter.java

```diff
@@ -68,7 +71,24 @@ public void write(EtlKey key, Object val) throws IOException, InterruptedExcepti
         //TODO: fix this logging message, should be logged once as a total count of old records skipped for each topic
         // for now, commenting this out
         //log.warn("Key's time: " + key + " is less than beginTime: " + beginTimeStamp);
-        ((Mapper.Context)context).getCounter("total", "skip-old").increment(1);
+        try {
```

Comment:
Doing reflection on every event is going to be very expensive. We need to cache the counter objects once at the beginning.